lua/lsp-status/statusline.lua

```diff
@@ -35,11 +35,47 @@ local function init(_, _config)
   _info = make_statusline_component('info')
 end
 
+local function get_lsp_progress()
+  local buf_messages = messages()
+  local msgs = {}
+
+  for _, msg in ipairs(buf_messages) do
+    local name = aliases[msg.name] or msg.name
+    local client_name = '[' .. name .. ']'
+    local contents
+    if msg.progress then
+      contents = msg.title
+      if msg.message then contents = contents .. ' ' .. msg.message end
+
+      if msg.percentage then contents = contents .. string.format(" (%.0f%%)", msg.percentage) end
```

Comment:
```suggestion
      if msg.percentage then contents = contents .. string.format(" (%.0f%%%%)", msg.percentage) end
```
Dreaded escape sequences: one for Lua and one for vim's statusline as that also has the percent symbol as a special character. Without these, the `%)` will never be rendered in the statusline.

The fix is pretty simple which can be included here, if not I can open another PR :)

Reply:
@dhruvmanila I actually noticed this after the commit that added the issue and then considered adding it here, but thought it might overstep the scope although given I've added the fix to my own config might as well add it here. 

Regarding the exact fix I don't think just adding the other percent sign there will work since I tried that I think since it's in a format expression it just seemed to break not sure how many are required. I think i'd just use `:gsub('%)', '%%)')` not sure. I'll try again and see. 

PS: if some reviewer comes by please feel free to review without this change. I'll add it if I get time before someone looks at this but otherwise would rather this just got reviewed.

Reply:
> Regarding the exact fix I don't think just adding the other percent sign there will work since I tried that I think since it's in a format expression it just seemed to break not sure how many are required. 

I have added it in my own config and it works with the two extra percent sign, but I do see that `:gsub` would be useful and make the code easy to read.

Also, thanks for the PR. I came here exactly looking for a message component :)

Reply:
Tried out your suggestion and it works fine @dhruvmanila so I've added it here 👍🏾

Reply:
It looks in the following way on my machine:
![изображение](https://user-images.githubusercontent.com/22453358/119043911-38a40780-b9c2-11eb-82b5-9c3f83578345.png)

Reply:
@Shatur95: Does that happen just with `clangd`? I haven't been able to reproduce it with other servers.

Reply:
Or maybe this is due to some extra escaping in `galaxyline`?

Reply:
Happens with other LSP servers for me. Should I escape it myself?

Reply:
@Shatur95 do you know how galaxy line draws the statusline? i.e. the actual way it structures the strings since this plugin now correctly returns `%%` which is the correct way to draw a `%` inside of a statusline.

> 	The option consists of printf style '%' items interspersed with
	normal text.  Each status line item is of the form:
	  %-0{minwid}.{maxwid}{item}
	All fields except the {item} are optional.  A single percent sign can
	be given as "%%".

I have no idea what galaxy line is doing though maybe it just constructs a string entirely and doesn't use vim's statusline syntax at all.

Reply:
You can test this behaviour by doing `set statusline=20%%` this will print `20%` in the statusline

Reply:
> it just constructs a string entirely and doesn't use vim's statusline syntax at all.

I think so... Will just escape it myself if it works correctly with default statusline.

Reply:
For what it's worth, even with `clangd` I've been unable to reproduce this with my bespoke statusline (and I don't think that particular servers should behave differently here). I think this might be galaxyline-specific, but I don't use galaxyline so I'm not sure.

Reply:
I think escaping `%` here is not the proper solution. If the expression is wrapped inside `{` stl item (NOT `{%`), then the resulting string's `%` is displayed as is. For example, `set stl=%{'%'}` displays `%` on the statusline. Similarly, the result of this function can be displayed on the stl by `set stl=%{luaeval('require(''lsp-status'').status_progress()')}`

Reply:
I think the statusline has too many different syntaxes so there isn't an easy way to get this right for everyone especially as statuslines are constructed differently.

That being said the solution of wrapping it in an item seems like it would have been the right call in hindsight. It will require users to know that they need to wrap the component though so that will need to be added to the README. I don't know how/or which statusline plugins this might break though but I think it's probably the most correct solution.

@tomtomjhj would you be up for raising your solution as a PR?

Reply:
Sure. I'll send a PR in few days.